app/web/features/search/ZoomControl.tsx

```diff
@@ -0,0 +1,75 @@
+import AddIcon from "@mui/icons-material/Add";
+import RemoveIcon from "@mui/icons-material/Remove";
+import { Box, IconButton, styled } from "@mui/material";
+import React from "react";
+import { MapRef } from "react-map-gl/maplibre";
+
+import { MAX_ZOOM_LEVEL, MIN_ZOOM_LEVEL } from "./utils/constants";
+
+interface ZoomControlProps {
+  mapRef: React.RefObject<MapRef>;
+  onZoomIn: (newZoom: number) => void;
+  onZoomOut: (newZoom: number) => void;
+  isZoomFromControlRef: React.MutableRefObject<boolean>;
+}
+
+const StyledIconButton = styled(IconButton)(({ theme }) => ({
+  "&:hover": {
+    borderRadius: 2,
+  },
+}));
+
+const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+const ZoomControl = ({
+  mapRef,
+  onZoomIn,
+  onZoomOut,
+  isZoomFromControlRef,
+}: ZoomControlProps) => {
+  const handleZoomIn = () => {
+    if (mapRef.current) {
+      delay(1000);
```

Comment:
why this delay? (out of curiosity)

Reply:
Ahhhh just removed it! I was trying to see if the zoom was messed up before because the smooth panning has a 2000ms duration. Ended up not being that though.